docs/cli.md

```diff
@@ -0,0 +1,667 @@
+# Hyperlane CLI
+
+The Hyperlane CLI is a command-line tool written in Typescript that facilitates common operations on Hyperlane, such as deploying the core contracts and/or warp routes to new chains.
+
+## Setup
+
+Node 16 or newer is required.
+
+## To install the CLI:
+
+```bash
+# Install the CLI globally
+npm install -g @hyperlane-xyz/cli
+```
+
+## To use the CLI:
+
+Just enter `hyperlane` in your terminal and you will see the available commands.
+
+```bash
+root@123: hyperlane
+
+Hyperlane CLI
+
+hyperlane <command>
+
+Commands:
+  hyperlane chains  View information about core Hyperlane chains
+  hyperlane config  Create or validate Hyperlane configs
+  hyperlane deploy  Permissionlessly deploy a Hyperlane contracts or extensions
+  hyperlane send    Send a test message or transfer
+  hyperlane status  Check status of a message
+
+Options:
+  --version  Show version number                                       [boolean]
+  --help     Show help                                                 [boolean]
+```
+
+The commands currently available are `chains`, `config`, `deploy`, `send`, and `status`.
+Each command has its own set of options and subcommands.
+
+Let's take a look at what each command does along with the options you get to use as well as the subcommands with their usual behavior.
+
+## Commands
+### chains
+
+The `chains` command is used to view information about core Hyperlane chains.
+
+```bash
+root@123: hyperlane chains
+```
+
+Thsi command pertains to getting information about the chains supported and the addresses of contracts as per the SDK.
+
+This would have the following output:
```

Comment:
```suggestion
The `chains` command is used to view information about core Hyperlane chains such as the set of supported chains and the addresses of deployed Hyperlane contracts on those chains.
```

imo no need to have a separate `hyperlane chains` example if it doesn't do anything itself

example below showing list/addresses is sufficient

ditto for other top-level commands you may write about